### DISCUSSION TITLE
What is buffer.tab_label?

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#154] [Q&A] [answered]
Not documented but shows in autocomplete.. always nil, unsettable (afaict) — just piques my curiosity  =)

**Comment by rgieseke** [comment, marked as answer]
See here: https://github.com/orbitalquark/textadept/blob/3ce1997b68b0454840870e35d84a4bb5feddfab3/core/ui.lua#L307

**Reply by metaleap** [reply]
Oh I see thanks, not a userland-customization thing then, I'll better leave it be :grin:

**Reply by rgieseke** [reply]
Well, i guess you could.
You need to have a function for all relevant events, here is something i used for shortening the buffer titles (i don't use tabs, but i guess you could do something similar there).

https://github.com/rgieseke/ta-common/blob/f1c43c74d11dd7d1a771577dbb08d178cf74c211/display_filename.lua

**Comment by orbitalquark** [comment]
It is documented: https://orbitalquark.github.io/textadept/api.html#buffer.tab_label, but you generally wouldn't modify it.
